src/main/java/org/myrobotlab/service/Agent.java

```diff
@@ -505,10 +509,10 @@ public synchronized void restart(String id) throws IOException, URISyntaxExcepti
     log.info("restarting process {}", id);
     kill(id); // FIXME - kill should include prepare to shutdown ...
     sleep(2000);
-    spawn(id);
+    spawnId(id);
   }
 
-  private void spawn(String id) {
+  public void spawnId(String id) {

```

Comment:
Why is this method public, and if we do want it to be public, we really need some javadoc on it.